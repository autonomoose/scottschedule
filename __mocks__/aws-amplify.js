export const { Hub } = jest.requireActual('aws-amplify');

const mockDefaultCurrentSession = {
    idToken: { payload: {'custom:tenant': 'test'} },
    accessToken: { payload: {
            username: 'mockuname',
            sub: 'mocksub'
}}};



export const Auth = {
    currentSession: jest.fn(() => Promise.resolve(mockDefaultCurrentSession)),
    currentAuthenticatedUser: jest.fn(() => Promise.resolve({
        username: 'testuser1',
        deleteUser: jest.fn(() => Promise.resolve('deleted')),
    })),
    userAttributes: jest.fn(() => Promise.resolve([
        {Name: 'sub', Value: 'test'},
        {Name: 'email_verified', Value: 'true'},
        {Name: 'email', Value: 'testuser1@test.com'},
        {Name: 'custom:tenant', Value: 'test'},
        {Name: 'phone_number_verified', Value: 'false'},
        {Name: 'phone_number', Value: '111'},
    ])),
    updateUserAttributes: jest.fn(() => Promise.resolve()),
    verifyCurrentUserAttribute: jest.fn(() => Promise.resolve()),
    verifyCurrentUserAttributeSubmit: jest.fn(() => Promise.resolve()),
    signIn: jest.fn(() => Promise.resolve()),
    signOut: jest.fn(() => Promise.resolve()),
    changePassword: jest.fn((currUsr, oldpass, newpass) => Promise.resolve('changed')),
};

export const Storage = {
    configure: jest.fn(() => Promise.resolve('origMockfile')),
    get: jest.fn(() => Promise.resolve('origMockfile')),
    put: jest.fn(() => Promise.resolve('origMockfile')),
    remove: jest.fn(() => Promise.resolve('removed origMockfile')),
    list: jest.fn(() => Promise.resolve([
        {key: "public/", size: 0},
        {key: "public/testlabel1", size: 23470},
        {key: "public/testlabel2", size: 22270},
    ]))
}

// used by layout before every authenticated page
export const mockAPIgqlGetCurrentUser = {
    'data': {
        'getCurrentUser': {
            adisable: null, agroups: null,
            nname: "user3test - legacy primary",
            perm: "standard", userid: "mockgraphql"
}}};
export const API = {
    post: jest.fn(() => Promise.resolve({'message': 'passed'})),
    graphql: jest.fn(() => Promise.resolve(mockAPIgqlGetCurrentUser)),
};

