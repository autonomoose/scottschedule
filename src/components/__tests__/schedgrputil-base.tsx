import React from "react";
import { render } from "@testing-library/react";

import { API } from 'aws-amplify';
import DisplaySchedGroup, { fetchSchedGroupsDB } from '../../components/schedgrputil';
jest.mock('aws-amplify');

const mockCallback = jest.fn();
const testSchedGroup = {
    name: 'testgrp',
    descr: 'test group',
    schedNames: [{
        begins: 'now',
        buttonName: 'test1',
        descr: 'test sched',
        schedName: 'testsched',
        schedTasks: [{evTaskId: 'testev'}],
        }]
};

const testDBgroupArgs = {
    begins: null,
    button: null,
    chain: null,
    descr: 'test schedules',
    evnames: 'default!args',
    sound: null,
    soundrepeat: null,
    warn: null,
};
const testDBschedArgs = {
    begins: 'now',
    button: 'test1',
    chain: null,
    descr: 'test sched',
    evnames: 'default!testsched!args',
    sound: '_default_',
    soundrepeat: '0',
    warn: '_none_',
};
const testDBevent = {
    begins: null,
    button: null,
    chain: null,
    descr: null,
    evnames: 'default!testsched!testev',
    sound: null,
    soundrepeat: null,
    warn: null,
};

const expectSchedNames = {
    begins: 'now',
    buttonName: 'test1',
    descr: 'test sched',
    schedName: 'testsched',
    schedTasks: [{evTaskId: 'testev'}],
};

const mytest = <DisplaySchedGroup group='default' groupSched={testSchedGroup} select={mockCallback} />
describe("schedgrputil - base DisplaySchedGroup event", () => {
  it("renders snapshot correctly", () => {
    const {container} = render(mytest);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("handles graphQL thrown error", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.reject('mockRejected')) as any;

      const schedGroups = await fetchSchedGroupsDB();
      expect(schedGroups).toStrictEqual({});
      API.graphql = prevAPIgraphql;
  });
  it("translates simple graphQL into schedgroups", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.resolve({'data':
        {'listSchedGroups': {items: [
          testDBgroupArgs,
          testDBschedArgs,
          testDBevent,
        ]}}
      })) as any;
      const schedGroups = await fetchSchedGroupsDB();

      expect(schedGroups).toStrictEqual({
        'default': {
          descr: 'test schedules',
          schedNames: [expectSchedNames],
        }
      });
      API.graphql = prevAPIgraphql;
  });
  it("translates graphQL schedgroups wo events", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.resolve({'data':
        {'listSchedGroups': {items: [
          testDBgroupArgs,
          testDBschedArgs,
        ]}}
      })) as any;
      const schedGroups = await fetchSchedGroupsDB();

      expect(schedGroups).toStrictEqual({
        'default': {
          descr: 'test schedules',
          schedNames: [{...expectSchedNames, schedTasks: [], }],
        }
      });
      API.graphql = prevAPIgraphql;
  });
  it("translates graphQL schedgroups wo optionals", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.resolve({'data':
        {'listSchedGroups': {items: [
          testDBgroupArgs,
          { evnames: 'default!testsched!args',},
        ]}}
      })) as any;
      const schedGroups = await fetchSchedGroupsDB();

      expect(schedGroups).toStrictEqual({
        'default': {
          descr: 'test schedules',
          schedNames: [{schedName: 'testsched', descr: '', schedTasks: [], }],
        }
      });
      API.graphql = prevAPIgraphql;
  });
  it("translates graphQL schedgroups w sound", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.resolve({'data':
        {'listSchedGroups': {items: [
          testDBgroupArgs,
          {...testDBschedArgs, sound: 'bigbell',},
          testDBevent,
        ]}}
      })) as any;
      const schedGroups = await fetchSchedGroupsDB();

      expect(schedGroups).toStrictEqual({
        'default': {
          descr: 'test schedules',
          schedNames: [{...expectSchedNames, sound: {name: 'bigbell',}, }],
        }
      });
      API.graphql = prevAPIgraphql;
  });
  it("translates graphQL schedgroups w repeated sound", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.resolve({'data':
        {'listSchedGroups': {items: [
          testDBgroupArgs,
          {...testDBschedArgs, sound: 'bigbell', soundrepeat: '2'},
          testDBevent,
        ]}}
      })) as any;
      const schedGroups = await fetchSchedGroupsDB();

      expect(schedGroups).toStrictEqual({
        'default': {
          descr: 'test schedules',
          schedNames: [{...expectSchedNames, sound: {name: 'bigbell', repeat: 2}, }],
        }
      });
      API.graphql = prevAPIgraphql;
  });
  it("translates graphQL schedgroups w warn", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.resolve({'data':
        {'listSchedGroups': {items: [
          testDBgroupArgs,
          {...testDBschedArgs, warn: '_default_',},
          testDBevent,
        ]}}
      })) as any;
      const schedGroups = await fetchSchedGroupsDB();

      expect(schedGroups).toStrictEqual({
        'default': {
          descr: 'test schedules',
          schedNames: [{...expectSchedNames, warn: {}, }],
        }
      });
      API.graphql = prevAPIgraphql;
  });
  it("translates graphQL schedgroups w chain", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.resolve({'data':
        {'listSchedGroups': {items: [
          testDBgroupArgs,
          {...testDBschedArgs, chain: 'testsched',},
          testDBevent,
        ]}}
      })) as any;
      const schedGroups = await fetchSchedGroupsDB();

      expect(schedGroups).toStrictEqual({
        'default': {
          descr: 'test schedules',
          schedNames: [{...expectSchedNames, chain: 'testsched', }],
        }
      });
      API.graphql = prevAPIgraphql;
  });

});


