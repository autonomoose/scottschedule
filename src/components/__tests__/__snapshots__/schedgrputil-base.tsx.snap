// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schedgrputil - base DisplaySchedGroup event renders snapshot correctly 1`] = `
<div
  class="MuiBox-root css-1w8v01d"
>
  <ul
    class="MuiList-root css-1mk9mw3-MuiList-root"
  >
    <div
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-1kyhxx1-MuiButtonBase-root-MuiListItem-root"
      role="button"
      tabindex="0"
    >
      default
       - 
      test group
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
    <ul
      class="MuiList-root MuiList-dense css-16vbaq4-MuiList-root"
    >
      <div
        class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding MuiListItem-button css-qyxyi3-MuiButtonBase-root-MuiListItem-root"
        role="button"
        tabindex="0"
      >
        testsched
         - 
        test sched
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </ul>
  </ul>
</div>
`;

exports[`schedgrputil - choice (ChoiceSchedGroup) event renders choice snapshot correctly 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-khqvv2-MuiFormControl-root-MuiTextField-root"
>
  <label
    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink="true"
    for="schedgroup"
    id="schedgroup-label"
  >
    Schedule Group
  </label>
  <div
    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    variant="outlined"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="schedgroup-label schedgroup"
      class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="schedgroup"
      role="button"
      tabindex="0"
    >
      default
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      data-testid="schedgroup"
      tabindex="-1"
      value="default"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-1in441m"
      >
        <span>
          Schedule Group
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`schedgrputil - connev (connect event) renders snapshot correctly 1`] = `
<div
  class="MuiBox-root css-13o7eu2"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="margin-top: 3px; max-width: 350px; min-width: 350px; flex: 1 1; box-shadow: -5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
  >
    <div
      class="MuiBox-root css-ibnnmb"
    >
      <form>
        <div
          class="MuiBox-root css-h9yv4t"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Connect Event
          </h6>
        </div>
        <div
          class="MuiBox-root css-1l4w6pd"
        >
          <div
            class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-viou3o-MuiAutocomplete-root"
            data-testid="taskid"
          >
            <div
              aria-invalid="false"
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="taskid"
                id="taskid-label"
              >
                Event Name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-invalid="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="taskid"
                  name="taskid"
                  role="combobox"
                  spellcheck="false"
                  type="text"
                  value=""
                />
                <div
                  class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Open"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                    tabindex="-1"
                    title="Open"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Event Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1csj9ua"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1ir7th6-MuiButtonBase-root-MuiButton-root"
            data-testid="evCancel"
            tabindex="0"
            type="button"
          >
            Cancel
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            data-testid="evRest"
            tabindex="0"
            type="button"
          >
            Reset
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root  css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            data-testid="evSave"
            tabindex="0"
            type="submit"
          >
            Save
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`schedgrputil - create group renders snapshot correctly 1`] = `
<div
  class="MuiBox-root css-2t3rs8"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="margin: 3px 0px 0px 0px; max-width: 350px; min-width: 350px; flex: 1 1; box-shadow: -5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
  >
    <div
      class="MuiBox-root css-0"
    >
      <form>
        <div
          class="MuiBox-root css-1j6jckw"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Add New Schedule Group
          </h6>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            X
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-o3707h"
        >
          <div
            aria-invalid="false"
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":rh:"
              id=":rh:-label"
            >
              Name
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="nameInput"
                id=":rh:"
                name="name"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-o3707h"
        >
          <label>
             Description
            <input
              aria-invalid="false"
              data-testid="descrInput"
              name="descr"
              size="30"
              type="text"
            />
          </label>
        </div>
        <div
          class="MuiBox-root css-1ol4ztf"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root Mui-disabled  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Reset
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root Mui-disabled  css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
